### PR TITLE
Export utils, generic _prepare_for_export

### PR DESCRIPTION
**Pull request by borisfom**
As per our design discussion.

### CODE REVIEW

File: nemo/utils/export_utils.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+default_1D_2D_replacements = {
+    "Conv1d": expand_Conv1D,
+    "BatchNorm1d": expand_BatchNorm1d,
+    "AdaptiveAvgPool1d": simple_expand(nn.AdaptiveAvgPool1d, nn.AdaptiveAvgPool2d),
+    "AvgPool1d": simple_expand(nn.AvgPool1d, nn.AvgPool2d),
+}
+
+
+def replace_modules(
```
Comment (okuchaiev) [inline nemo/utils/export_utils.py:126]: needs docstring

File: nemo/utils/export_utils.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        if m_type in expansions:
+            swapped = expansions[m_type](m)
+            if swapped:
+                mapping[name] = swapped
+    logging.warning(f"Swapped {len(mapping)} modules")
+    swap_module(model, mapping)
+    return model
+
+
+def replace_for_export(
```
Comment (okuchaiev) [inline nemo/utils/export_utils.py:141]: why this function is needed? All it does is call to replace_modules.
Also, it's "expansions" argument is not used

File: nemo/collections/asr/modules/conv_asr.py [hunk 3]
```diff
@@ -239,6 +239,16 @@ def input_example(self):
         input_example = torch.randn(bs, self._feat_in, seq).to(next(self.parameters()).device)
         return tuple([input_example])
 
+    def _prepare_for_export(self):
+        m_count = 0
+        for m in self.modules():
+            if type(m).__name__ == "MaskedConv1d":
```
[the remark this answers is not in the document]
Comment (jekirl) [reply nemo/collections/asr/modules/conv_asr.py:245]: I would agree with this as well for purposes of type checking/soundness, the string matching by type name came from the original implementation. 

Otherwise even worse would be scenario where the class name does not change but the name of `use_mask` changes  such that the conditional would be triggered but the `setattr(m, "use_mask", False)` would throw an assertion

File: nemo/utils/export_utils.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    for _ in range(2):
+        sample_input = torch.rand(1, conv1d.in_channels, 256)
+        close = conv1d(sample_input).mean() - conv2d(sample_input.unsqueeze(-1)).squeeze().mean()
+        if close.abs() > 1e-6:
+            raise ValueError("Unable to expand Conv1D to Conv2D")
+    return conv2d
+
+
+def expand_BatchNorm1d(bn1d: nn.Module) -> Optional[nn.BatchNorm2d]:
+    if not isinstance(bn1d, nn.BatchNorm1d):
```
[the remark this answers is not in the document]
Comment (jekirl) [reply nemo/utils/export_utils.py:75]: Strictly speaking this isn't required for same reasons as `expand_Conv1D`. The Onnx export is identical for BatchNorm1d and BatchNorm2d, the only reason this is required is due  to the way torch implemented `_check_input_dim` on `_BatchNorm`. Doing this expansion is just to maintain sanity within pytorch when printing out the graph. You could just as easily make the check a noop like this:

```python
    def _check_input_dim(*args):
        return True
    setattr(b1, "_check_input_dim", _check_input_dim)


```

https://github.com/pytorch/pytorch/blob/51861cc9b19d9c483598e39932661822a826d3a2/torch/nn/modules/batchnorm.py#L275

**Review by okuchaiev — DISMISSED** [review]
